python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,533 @@
+    pipeline_config : Dict[GraphExecutorFactoryModule, Dict[str, Any]]
+        modules and modules dependency configuration informaiton.
+    """
+
+    def __init__(self, pipe_mod_config):
+        self.pipeline_mods = pipe_mod_config.pipeline_mods
+        self.mod_config = pipe_mod_config.mods_config
+        mods, config = self.graph_executor_create(self.pipeline_mods, self.mod_config)
+        assert (
+            pipeline_executor_enabled()
+        ), "Pipeline executor is not enabled. Please \
+              re-build TVM with USE_PIPELINE_EXECUTOR=ON"
+        pipeline_create = tvm._ffi.get_global_func(
+            "tvm.pipeline_executor.create", allow_missing=False
+        )
+        assert pipeline_create
+        module = pipeline_create(mods, config)
+
+        self.module_ = module
+
+    def graph_executor_create(self, pipeline_mods, mod_config):
+        """Create graph_executor list and return text format configuration.
+
+        Parameters
+        ----------
+        pipeline_mods : List[GraphExecutorFactoryModule]
+          list of GraphExecutorFactoryModule
+
+        mod_config : Dict[str, Any]
+            modules dependency configuration informaiton.
+
+        Returns
+        -------
+        mods : List[Module]
+            Module list.
+
+        mod_config : str
+            mods configuration
+        """
+
+        mods = []
+        for pipeline_mod in pipeline_mods:
+            mod = graph_executor.GraphModule(
+                pipeline_mod["default"](pipeline_mods[pipeline_mod]["dev"])
+            )
+            mods.append(mod.module)
+
+        return mods, json.dumps(mod_config)
+
+
+class PipelineConfig(object):
+    """The wrapper of each module to be pipelined. The wrapper mainly includes the
+    module itself as well as the binding that represents the connections of this
+    module's inputs and outputs to other modules.
+    """
+
+    class Binding:
+        """The class that use to storage module connection information.
+        The binding can be either "input" or "output".
+
+        Parameters
+        ----------
+        owner : ModuleWrapper
+            The class that own this interface, in such class there are
+            Module information like idx, module name
+
+        io_type : str
+            The type of this binding. It can be either "input" or "output".
+
+        name : str/integer
+            Binding name, for input it is string such as "data0";
+            for output it is the idx integer such as 0.
+        """
+
+        def __init__(self, owner, stype, name, data_type=None):
+            self.io_owner = owner
+            self.io_type = stype
+            self.name = str(name)
+            # These item that have dependency relation with self
+            self.bindings = []
+            # The item that self depend
+            self.parents = []
+
+            self.data_type = data_type
+
+        def get_name(self):
+            """Get owner name and self name"""
```

Comment:
what is self name

Reply:
changed into "Return this interface name and name of owner who own this interface."